lib/puppet/file_serving/terminus_selector.rb

```diff
@@ -21,6 +26,8 @@ def select(request)
       else
         Puppet[:default_file_terminus]
       end
+   when "http","https"
+     :http
```

Comment:
Oh, maybe this is related to your comments about the differences between `agent` and `apply`?